examples/oft-aptos/tasks/evm/wire-evm.ts

```diff
@@ -0,0 +1,145 @@
+import { executeTransactions } from './wire/transactionExecutor'
+
+if (!process.env.PRIVATE_KEY) {
+    console.error('PRIVATE_KEY environment variable is not set.')
+    process.exit(1)
+}
+
+// @todo Fetch this from the config instead of hardcoding.
+const EID_APTOS = EndpointId.APTOS_V2_SANDBOX
+
+/**
+ * @description Handles wiring of EVM contracts with the Aptos OApp
+ * @dev Creates ethers's populated transactions for the various transaction types (setPeer, setDelegate, setEnforcedOptions, setSendLibrary, setReceiveLibrary, setReceiveLibraryTimeout). It then simulates them on a forked network before executing
+ */
+async function main() {
+    // @todo grow connectionsToWire by taking in non-evm connections instead of only APTOS.
+    const connectionsToWire = getConfigConnections('to', EID_APTOS)
+
+    const accountConfigs = getHHAccountConfig()
+    const networks = createEidToNetworkMapping()
+    const rpcUrls = createEidToNetworkMapping('url')
+
+    // Build a Transaction mapping for each type of transaction. It is further indexed by the eid.
+    const TxTypeEidMapping: TxEidMapping = {
+        setPeer: {},
+        setDelegate: {},
+        setEnforcedOptions: {},
+        setSendLibrary: {},
+        setReceiveLibrary: {},
+        setReceiveLibraryTimeout: {},
+        sendConfig: {},
+        receiveConfig: {},
+    }
+
+    // Indexed by the eid it contains information about the contract, provider, and configuration of the account and oapp.
+    const contractMetaData: ContractMetadataMapping = {}
+
+    // @todo Fetch this from the config instead of hardcoding.
+    // @todo Use this as a primary key for NonEvmOAppWiring in the following code
+    const APTOS_OAPP_ADDRESS = '0x8401fa82eea1096b32fd39207889152f947d78de1b65976109493584636622a8'
+    const nonEvmOapp: NonEvmOAppMetadata = {
+        address: APTOS_OAPP_ADDRESS,
+        eid: EID_APTOS,
+        rpc: rpcUrls[EID_APTOS],
+    }
+
+    /*
+     * Looping through the connections we build out the contractMetaData and TxTypeEidMapping by reading from the deployment files.
+     * contractMetaData contains ethers Contract objects for the OApp and EndpointV2 contracts.
+     */
+    for (const conn of connectionsToWire) {
+        const fromEid = conn.from.eid as eid
+        const fromNetwork = networks[fromEid]
+
+        const provider = new ethers.providers.JsonRpcProvider(rpcUrls[fromEid])
+        const signer = new ethers.Wallet(process.env.PRIVATE_KEY, provider)
+
+        const OAppDeploymentPath = `deployments/${fromNetwork}/${conn.from.contractName}.json`
+        const OAppDeploymentData = JSON.parse(fs.readFileSync(OAppDeploymentPath, 'utf8'))
+
+        const { address: oappAddress, abi: oappAbi, bytecode: oappBytecode } = OAppDeploymentData
+        const OAppFactory = new ContractFactory(oappAbi, oappBytecode, signer)
+
+        const EndpointV2DeploymentPath = `deployments/${fromNetwork}/EndpointV2.json`
```

Comment:
Just make sure you're getting this from `@layerzerolabs/toolbox-hardhat`. This package is pulling in the `EndpointV2.json` for every network via `@layerzerolabs/lz-evm-sdk-v2`. 

@shankars99

Reply:
Yep. That was a todo on the cleanup